package.json

```diff
@@ -18,86 +18,127 @@
+    "@babel/core": "^7.28.0",
     "@babel/eslint-parser": "^7.28.0",
-    "@babel/helper-function-name": "^7.24.7",
-    "@babel/plugin-proposal-decorators": "^7.28.0",
-    "@ember/optional-features": "^2.2.0",
+    "@babel/plugin-transform-typescript": "^7.28.0",
+    "@babel/runtime": "^7.27.6",
+    "@ember/app-tsconfig": "^1.0.3",
+    "@ember/library-tsconfig": "^1.1.3",
     "@ember/string": "^4.0.1",
     "@ember/test-helpers": "^5.2.2",
-    "@embroider/test-setup": "^4.0.0",
-    "@glimmer/component": "^1.1.2",
-    "@glimmer/tracking": "^1.1.2",
-    "broccoli-asset-rev": "^3.0.0",
+    "@embroider/addon-dev": "^8.1.0",
+    "@embroider/compat": "^4.1.0",
+    "@embroider/core": "^4.1.1",
+    "@embroider/macros": "^1.18.0",
+    "@embroider/vite": "^1.1.5",
+    "@eslint/js": "^9.31.0",
+    "@glimmer/component": "^2.0.0",
+    "@glint/core": "^2.0.0-alpha.3",
+    "@glint/environment-ember-loose": "^2.0.0-alpha.3",
+    "@glint/environment-ember-template-imports": "^2.0.0-alpha.3",
+    "@glint/template": "^1.6.0-alpha.2",
+    "@glint/tsserver-plugin": "^2.0.0-alpha.3",
+    "@rollup/plugin-babel": "^6.0.4",
+    "@types/qunit": "^2.19.12",
+    "babel-plugin-ember-template-compilation": "^3.0.0",
     "concurrently": "^9.2.0",
-    "ember-cli": "~5.12.0",
-    "ember-cli-clean-css": "^3.0.0",
-    "ember-cli-dependency-checker": "^3.3.3",
-    "ember-cli-inject-live-reload": "^2.1.0",
-    "ember-cli-sri": "^2.1.1",
-    "ember-cli-terser": "^4.0.2",
-    "ember-exam": "^9.1.0",
-    "ember-load-initializers": "^3.0.1",
-    "ember-page-title": "^8.2.4",
-    "ember-qunit": "^8.1.1",
-    "ember-resolver": "^12.0.1",
-    "ember-source": "~5.12.0",
-    "ember-source-channel-url": "^3.0.0",
-    "ember-template-lint": "^6.1.0",
-    "ember-try": "^3.0.0",
-    "eslint": "^8.57.1",
-    "eslint-config-prettier": "^9.1.0",
+    "ember-qunit": "^9.0.3",
+    "ember-resolver": "^13.1.1",
+    "ember-source": "^6.5.0",
+    "ember-template-lint": "^7.9.1",
+    "eslint": "^9.31.0",
+    "eslint-config-prettier": "^10.1.5",
     "eslint-plugin-ember": "^12.6.0",
+    "eslint-plugin-import": "^2.32.0",
     "eslint-plugin-n": "^17.21.0",
-    "eslint-plugin-prettier": "^5.5.1",
-    "eslint-plugin-qunit": "^8.2.4",
-    "loader.js": "^4.7.0",
+    "globals": "^16.3.0",
     "prettier": "^3.6.2",
+    "prettier-plugin-ember-template-tag": "^2.0.6",
     "qunit": "^2.24.1",
     "qunit-dom": "^3.4.0",
-    "release-plan": "^0.16.0",
-    "stylelint": "^15.11.0",
-    "stylelint-config-standard": "^34.0.0",
-    "stylelint-prettier": "^4.1.0",
-    "webpack": "^5.100.2"
+    "release-plan": "^0.17.0",
+    "rollup": "^4.45.1",
+    "testem": "^3.16.0",
+    "typescript": "~5.8.3",
+    "typescript-eslint": "^8.37.0",
+    "vite": "^6.3.5"
   },
   "peerDependencies": {
-    "@ember/string": ">= 3.1.1",
-    "ember-source": ">= 3.28.0"
-  },
-  "engines": {
-    "node": ">= 20",
-    "pnpm": ">= 10"
+    "@ember/string": ">= 3.1.1"
```

Comment:
do we still need this? this has been a pain for most people to deal with so if we can inline the code that is actually being used form `@ember/string` or something that would be great. At the very least we should be making this a dependency on ember/string and the latest version, now that we're a v2 addon we don't need to worry about the strange broccoli mixing (I think)